### PR TITLE
use lc_solr_sortable repo for processing call number sort vals

### PR DESCRIPTION
**Pull request by quoideneuf**

### CODE REVIEW

**Review by dkinzer — APPROVED** [review]

**Comment by dkinzer** [comment]
@quoideneuf looks good, but I'm curious; why do we process call numbers in the same way both at query time (from the app) and at ingest time?

**Comment by quoideneuf** [comment]
> @quoideneuf looks good, but I'm curious; why do we process call numbers in the same way both at query time (from the app) and at ingest time?

@dkinzer The main reason is that unprocessed call numbers don't sort very well using the default `alphaOnlySort` field type in solr. Likewise, I think if we tried to use the unprocessed numbers to generate a range, we'd get incorrect results (I am fairly certain that a field that isn't sortable won't support a range query).

**Comment by dkinzer** [comment]
@quoideneuf I mean, once we processed at index time and make them sortable wont they already be processed app side?  And, can't we use the already processed form app side?

**Comment by dkinzer** [comment]
Actually, I think I get it.  We would need to update our lc display field to be a JSON object and include both the sortable value and the display value... And then we would need to update the app to work with the new format.  So I guess in terms of changes and complexity your overall approach makes more sense.  Thanks.

File: lib/cob_index/macros/custom.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@
 require "library_stdnums"
 require "active_support/core_ext/object/blank"
 require "time"
+require 'lc_solr_sortable'
```
Comment (quoideneuf) [inline lib/cob_index/macros/custom.rb:7]:
```suggestion
require "lc_solr_sortable"
```

**Comment by quoideneuf** [comment]
> Actually, I think I get it. We would need to update our lc display field to be a JSON object and include both the sortable value and the display value... And then we would need to update the app to work with the new format. So I guess in terms of changes and complexity your overall approach makes more sense. Thanks.

That is right. But also, because LC numbers are segmented, we might need to generate a value for a boundary like "A" that isn't an actual call number.

**Pull request merged**